### PR TITLE
Arcamdev

### PR DESCRIPTION
**Pull request by PicassoCT**
Refactored the commits - removed the flags.

### CODE REVIEW

File: rts/Game/Camera/ARController.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+void CARController::MouseWheelMove(float move)
+{
+
+	Update();
+}
+
+void CARController::Update()
+{
+	pos.x = pos.x;
```
Comment (ashdnazg) [inline rts/Game/Camera/ARController.cpp]: huh?

**Comment by ashdnazg** [comment]
Can you please explain how this controller has anything to do with AR?
Also seems like screenDivideRatio and mouseScale aren't really used?

**Comment by PicassoCT** [comment]
All Spring Cameras have Boundaries, as in - you can not override the Position in Lua outside these bounds.
You are correct that the ScreenDivideRatio is currently not used.
This  is a first version of the camera, to be used with only One Camera rendering one screen.

The final version will consist of two cameras rendering from a slightly diffrent angle.
Im also planning to reduce/upping screen-resolution depending on the cellphone, but that is also far away at the moment.

Important is that the camera has no hardcoded boundaries.

File: rts/Game/Camera/ARController.cpp (hunk 1 above)
Comment (PicassoCT) [reply rts/Game/Camera/ARController.cpp]: Sorry, that is a mistake .

**Comment by ashdnazg** [comment]
If they'll only be needed later, add them later...

File: rts/Game/CameraHandler.cpp [hunk 2]
```diff
@@ -169,15 +172,18 @@ void CCameraHandler::InitControllers()
...
-	camControllers[CAMERA_MODE_ROTOVERHEAD] = new (camControllerMem[CAMERA_MODE_ROTOVERHEAD]) CRotOverheadController();
-	camControllers[CAMERA_MODE_FREE       ] = new (camControllerMem[CAMERA_MODE_FREE       ])        CFreeController();
-	camControllers[CAMERA_MODE_OVERVIEW   ] = new (camControllerMem[CAMERA_MODE_OVERVIEW   ])    COverviewController();
+	camControllers[CAMERA_MODE_FIRSTPERSON] = new (camControllerMem[CAMERA_MODE_FIRSTPERSON])   CFPSController();
+	camControllers[CAMERA_MODE_OVERHEAD   ] = new (camControllerMem[CAMERA_MODE_OVERHEAD   ])   COverheadController();
+	camControllers[CAMERA_MODE_SPRING     ] = new (camControllerMem[CAMERA_MODE_SPRING     ])   CSpringController();
+	camControllers[CAMERA_MODE_ROTOVERHEAD] = new (camControllerMem[CAMERA_MODE_ROTOVERHEAD]) 	CRotOverheadController();
+	camControllers[CAMERA_MODE_FREE       ] = new (camControllerMem[CAMERA_MODE_FREE       ])   CFreeController();
+	camControllers[CAMERA_MODE_OVERVIEW   ] = new (camControllerMem[CAMERA_MODE_OVERVIEW   ])   COverviewController();
+	camControllers[CAMERA_MODE_AR  		  ]	= new (camControllerMem[CAMERA_MODE_AR   	   ])   CARController();
```
Comment (ashdnazg) [inline rts/Game/CameraHandler.cpp]: some whitespace issues

File: rts/Game/CameraHandler.cpp [hunk 3]
```diff
@@ -169,15 +172,18 @@ void CCameraHandler::InitControllers()
 	static_assert(sizeof(     CSpringController) <= sizeof(camControllerMem[CAMERA_MODE_SPRING     ]), "");
 	static_assert(sizeof(CRotOverheadController) <= sizeof(camControllerMem[CAMERA_MODE_ROTOVERHEAD]), "");
 	static_assert(sizeof(       CFreeController) <= sizeof(camControllerMem[CAMERA_MODE_FREE       ]), "");
-	static_assert(sizeof(   COverviewController) <= sizeof(camControllerMem[CAMERA_MODE_OVERVIEW   ]), "");
+	static_assert(sizeof(   COverviewController) <= sizeof(camControllerMem[CAMERA_MODE_AR  ]), "");
+	static_assert(sizeof(   CARController) <= sizeof(camControllerMem[CAMERA_MODE_OVERVIEW   ]), "");
```
Comment (ashdnazg) [inline rts/Game/CameraHandler.cpp]: whitespace alignment here as well

**Comment by rtri** [comment]
Remove all empty function bodies, put them in the header on a single line. Do the same with `SwitchFrom`.

Restore the original right-justified *Controller formatting from https://github.com/spring/spring/blob/develop/rts/Game/CameraHandler.cpp#L175.

For bonus points, add override qualifiers.

File: rts/Game/CameraHandler.cpp (hunk 3 above)
Comment (PicassoCT) [reply rts/Game/CameraHandler.cpp]: And a switcheroo..
fixed

File: rts/Game/CameraHandler.cpp [hunk 4]
```diff
@@ -172,17 +172,17 @@ void CCameraHandler::InitControllers()
...
 	camControllers[CAMERA_MODE_FIRSTPERSON] = new (camControllerMem[CAMERA_MODE_FIRSTPERSON])   CFPSController();
 	camControllers[CAMERA_MODE_OVERHEAD   ] = new (camControllerMem[CAMERA_MODE_OVERHEAD   ])   COverheadController();
-	camControllers[CAMERA_MODE_SPRING     ] = new (camControllerMem[CAMERA_MODE_SPRING     ])   CSpringController();
+	camControllers[CAMERA_MODE_SPRING	  ] = new (camControllerMem[CAMERA_MODE_SPRING     ])   CSpringController();
 	camControllers[CAMERA_MODE_ROTOVERHEAD] = new (camControllerMem[CAMERA_MODE_ROTOVERHEAD])   CRotOverheadController();
-	camControllers[CAMERA_MODE_FREE       ] = new (camControllerMem[CAMERA_MODE_FREE       ])   CFreeController();
+	camControllers[CAMERA_MODE_FREE 	  ] = new (camControllerMem[CAMERA_MODE_FREE       ])   CFreeController();
 	camControllers[CAMERA_MODE_OVERVIEW   ] = new (camControllerMem[CAMERA_MODE_OVERVIEW   ])   COverviewController();
-	camControllers[CAMERA_MODE_AR         ]	= new (camControllerMem[CAMERA_MODE_AR         ])   CARController();
+	camControllers[CAMERA_MODE_AR   	  ]	= new (camControllerMem[CAMERA_MODE_AR         ])   CARController();
```
Comment (ashdnazg) [inline rts/Game/CameraHandler.cpp]: alignment is only spaces

**Comment by sprunk** [comment]
>All Spring Cameras have Boundaries, as in - you can not override the Position in Lua outside these bounds.

https://github.com/spring/spring/blob/0ae2035c200add1fb2f67ca46e075fb94a7b42e4/rts/Game/Camera/FPSController.cpp#L21

**Pull request closed without merging**